### PR TITLE
SSR sites only force deploy the SSR function

### PR DESCRIPTION
**Pull request by inlined**
Steps to reproduce:

1. Create a function
2. Create the sample Next.js app
3. Call `firebase deploy --only hosting`

Expected: hosting and `firebase-frameworks-${site}` function are deployed
Actual: All hosting and functions are deployed.

### CODE REVIEW

File: src/frameworks/index.ts [hunk 1]
```diff
@@ -411,7 +411,12 @@ You can link a Web app to a Hosting site here https://console.firebase.google.co
         },
       ]);
 
-      if (!targetNames.includes("functions")) targetNames.unshift("functions");
+      if (!targetNames.includes("functions")) {
+        targetNames.unshift("functions");
+        if (options.only) {
```
Comment (bkendall) [inline src/frameworks/index.ts:416]: May want to make sure `functions` isn't included already in `only`. I'm not sure what the ramifications would be off hand.

**Review by bkendall — APPROVED** [review]

File: src/frameworks/index.ts [hunk 2]
```diff
@@ -411,7 +411,12 @@ You can link a Web app to a Hosting site here https://console.firebase.google.co
...
+          options.only = `${options.only},functions:firebase-frameworks-${site}`;
```
Comment (jamesdaniels) [inline src/frameworks/index.ts:417]: Can we defensively DRY up the codebase name? Incase we need to add a transformation or something down the road.

**Review by TheIronDev — APPROVED** [review]

File: src/frameworks/index.ts [hunk 3]
```diff
@@ -24,8 +24,10 @@ import { FirebaseError } from "../error";
 import { requireHostingSite } from "../requireHostingSite";
 import { HostingRewrites } from "../firebaseConfig";
 import * as experiments from "../experiments";
+import { ensureTargeted } from "../functions/ensureTargeted";
 import { implicitInit } from "../hosting/implicitInit";
 
+
```
Comment (TheIronDev) [inline src/frameworks/index.ts:30]: Empty space

**Review by TheIronDev — APPROVED** [review]
LGTM

**Pull request merged**